sympy/polys/matrices/domainmatrix.py

```diff
@@ -770,6 +781,36 @@ def matmul(A, B):
         A._check('*', B, A.shape[1], B.shape[0])
         return A.from_rep(A.rep.matmul(B.rep))
 
+    def scalarmul(A, lamda):
+        if lamda.element == lamda.domain.zero:
+            m, n = A.shape
+            return DomainMatrix([[lamda.domain.zero]*n]*m, (m, n), A.domain)
+        if lamda.element == lamda.domain.one:
+            return A
+
+        return A.mul(lamda.element)
+
+    def __truediv__(A, lamda):
+        """ Method for Scalar Divison"""
+        if isinstance(lamda, int):
+            if lamda == S.Zero:
+                raise ZeroDivisionError
+            elif lamda == S.One:
+                return A.to_field()
+            else:
+                lamda = DomainScalar(ZZ(lamda), ZZ)
```

Comment:
This doesn't need the different cases. It should just convert to `DomainScalar` and let the code below handle zero and one.

Reply:
Sure, I added so that we can save converting to DomainScalar for int(0) or int(1), but it's lengthy.
Will do the same as before.
Thanks.